File: PWR015/example-omp.c
// PWR015: Avoid copying unnecessary array elements to the GPU
// https://www.codee.com/knowledge/pwr015

void foo() {
    int A[100], B[100], sum[100];
    #pragma omp target map(to: A[0:100], B[0:100]) map(from: sum[0:100])
    #pragma omp parallel for
    for (int i = 0; i < 50; i++) {
        sum[i] = A[i] + B[i];
    }
}

File: PWR015/example-omp.f90
! PWR015: Avoid copying unnecessary array elements to the GPU
! https://www.codee.com/knowledge/pwr015

subroutine example(a, c)
    integer, intent(inout) :: a(100), c(100)
    integer :: i
    !$omp target teams distribute parallel do schedule(auto) shared(A, B) map(to: a(0:100)) map(tofrom: c(0:100))
    do i = 1, 50
        c(i) = c(i) + a(i)
    end do
end subroutine
